ospfd/ospf_vty.c

```diff
@@ -12301,6 +12331,9 @@ static int config_write_ospf_external_aggregator(struct vty *vty,
 			if (aggr->tag)
 				vty_out(vty, " tag %u ", aggr->tag);
 
+			if (aggr->metric != -1)
+				vty_out(vty, " metric %d ", aggr->metric);
```

Comment:
`%d` can print a negative metric here (provided it's big enough).

IMO it would be better to add a new flag to `ospf_external_aggr_rt` instead of using `-1` to signify the absence of a metric.

Reply:
This is for show running config , normally default values will not be printed in the show running configs right ? so  here the default value of metric  is -1 so adding to running config other than -1.

Reply:
There's a redundant space at the end, please remove it.

Reply:
addressed